app/models/package_manager/maven.rb

```diff
@@ -18,15 +19,41 @@ class Maven < Base
     }.freeze
 
     def self.package_link(project, version = nil)
+      if version
+        db_version = project.versions.find_by(number: version)
+        if db_version.repository_sources.include?(REPOSITORY_SOURCE_NAME)
+          return MavenUrl.from_name(project.name, repository_base).search(version)
+        else
+          repository_source = db_version.repository_sources.first
+          return "MavenManager::#{repository_source}".constantize.package_link(project, version)
+        end
+      end
+
       MavenUrl.from_name(project.name, repository_base).search(version)
     end
 
     def self.download_url(name, version = nil)
+      if version
+        project = Project.find_by(name: name, platform: self.name.demodulize)
+        db_version = project.versions.find_by(number: version)
+        if db_version.repository_sources.include?(REPOSITORY_SOURCE_NAME)
+          return MavenUrl.from_name(name, repository_base).jar(version)
+        else
+          repository_source = db_version.repository_sources.first
+          return "MavenManager::#{repository_source}".constantize.download_url(name, version)
+        end
+      end
+
       MavenUrl.from_name(name, repository_base).jar(version)
     end
 
     def self.check_status_url(project)
-      MavenUrl.from_name(project.name, repository_base).base
+      sources = project.versions.flat_map(&:repository_sources).uniq
+      if sources.include?(REPOSITORY_SOURCE_NAME) 
+        MavenUrl.from_name(project.name, repository_base).base
+      else
+        "MavenManager::#{sources.first}".constantize.check_status_url(project)
```

Comment:
Could this get a little simpler if extracted out maven central and treated it exactly how we do the other repositories.

Reply:
That's not a bad idea. I'm struggling with the selection of the source in here. In my mind, I wanted Maven Central to be the default, but it doesn't need to be. The only perceived "main" source is from the link shown on the Maven section web page which is driven by https://github.com/librariesio/libraries.io/blob/af2c4dbe4c6f4626c12c096b9d569c0188397c0a/app/models/package_manager/maven.rb#L11

Reply:
Def gets rid of all the ` if sources.include?(REPOSITORY_SOURCE_NAME) ` kinda stuff